dll/win32/shell32/CFolderOptions.cpp

```diff
@@ -100,6 +103,45 @@ HRESULT STDMETHODCALLTYPE CFolderOptions::SetSite(IUnknown *pUnkSite)
 
 HRESULT STDMETHODCALLTYPE CFolderOptions::GetSite(REFIID riid, void **ppvSite)
 {
-    return m_pSite->QueryInterface(riid, ppvSite);
+    return m_pSite ? m_pSite->QueryInterface(riid, ppvSite) : E_FAIL;
 }
 
+/*************************************************************************
+ * FolderOptions helper methods
+ */
+HRESULT CFolderOptions::HandleDefFolderSettings(int Action)
+{
+    IBrowserService2*bs2;
+    HRESULT hr = IUnknown_QueryService(m_pSite, SID_SShellBrowser, IID_PPV_ARG(IBrowserService2, &bs2));
+    if (SUCCEEDED(hr))
+    {
+        if (Action == DFSA_APPLY)
+        {
+            hr = bs2->SetAsDefFolderSettings();
+        }
+        else if (Action == DFSA_RESET)
+        {
+            // There does not seem to be a method in IBrowserService2 for this
+            IUnknown_Exec(bs2, CGID_DefView, DVCMDID_RESET_DEFAULTFOLDER_SETTINGS, OLECMDEXECOPT_DODEFAULT, NULL, NULL);
+        }
+        else
```

Comment:
I would delete this else statement personally, but any compiler worth its salt is going to toss this out.